jwcrypto/jwe.py

```diff
@@ -253,17 +255,21 @@ def add_recipient(self, key, header=None):
 
         if 'recipients' in self.objects:
             self.objects['recipients'].append(rec)
-        elif 'encrypted_key' in self.objects or 'header' in self.objects:
-            self.objects['recipients'] = []
-            n = {}
-            if 'encrypted_key' in self.objects:
-                n['encrypted_key'] = self.objects.pop('encrypted_key')
-            if 'header' in self.objects:
-                n['header'] = self.objects.pop('header')
-            self.objects['recipients'].append(n)
-            self.objects['recipients'].append(rec)
+        elif self.flattened:
+            if 'encrypted_key' in self.objects or 'header' in self.objects:
+                self.objects['recipients'] = []
+                n = {}
+                if 'encrypted_key' in self.objects:
+                    n['encrypted_key'] = self.objects.pop('encrypted_key')
+                if 'header' in self.objects:
+                    n['header'] = self.objects.pop('header')
+                self.objects['recipients'].append(n)
+                self.objects['recipients'].append(rec)
+            else:
+                self.objects.update(rec)
         else:
-            self.objects.update(rec)
+            self.objects['recipients'] = [rec]
+
```

Comment:
remove this line to make pep8 happy and I'll merge.

Reply:
ah you just did push a new commit for this :)

Reply:
Yes I missed this the first time, sorry for the back and forth!